cmd/secrets-provider/main.go

```diff
@@ -128,3 +126,21 @@ func configureLogLevel() {
 		log.Warn(messages.CSPFK001W, val, validVal)
 	}
 }
+
+func validateContainerMode(containerMode string) {
+	validContainerModes := []string{
+		"init",
+		"application",
+	}
+
+	isValidContainerMode := false
+	for _, validContainerModeType := range validContainerModes {
+		if containerMode == validContainerModeType {
+			isValidContainerMode = true
+		}
+	}
+
+	if !isValidContainerMode {
+		printErrorAndExit(fmt.Sprintf(messages.CSPFK007E, containerMode, validContainerModes))
+	}
```

Comment:
LOVE this code :)

Reply:
I'm 😱 !
Just found out there's no way to find an element in an array in Go!
Can't believe it...

Reply:
yeah i had the same shock around 2 weeks ago. crazy that they don't have a built-in method